xtask/src/deny.rs

```diff
@@ -0,0 +1,55 @@
+//! Run `cargo deny`
+//!
+//! Also installs the subcommand if it is not already installed.
+
+use std::process::Command;
+
+use super::DynError;
+
+pub fn deny(args: &[&str]) -> Result<(), DynError> {
+    if !args.is_empty() {
+        eprintln!("Warning: arguments ignored: {args:?}");
+    }
+    install_cargo_deny()?;
+    let targets = [
+        "aarch64-apple-darwin",
+        "aarch64-linux-android",
+        "i686-pc-windows-gnu",
+        "i686-pc-windows-msvc",
+        "i686-unknown-linux-gnu",
+        "wasm32-unknown-unknown",
+        "x86_64-apple-darwin",
+        "x86_64-pc-windows-gnu",
+        "x86_64-pc-windows-msvc",
+        "x86_64-unknown-linux-gnu",
+        "x86_64-unknown-linux-musl",
+        "x86_64-unknown-redox",
+    ];
+    for target in targets {
+        let status = Command::new("cargo")
+            .args([
+                "deny",
+                "--all-features",
+                "--log-level",
+                "error",
+                "--target",
+                target,
+                "check",
+            ])
+            .status()?;
+        if !status.success() {
+            return Err(status.to_string().into());
+        }
+    }
+    Ok(())
+}
+
+fn install_cargo_deny() -> Result<(), DynError> {
```

Comment:
Should we prompt the user to see if they are ok with installing packages? The shell scripts don't do that, but it seems like a good idea. There is an example in miri: <https://github.com/rust-lang/miri/blob/dba35d2be72f4b78343d1a0f0b4737306f310672/cargo-miri/src/util.rs#L181-L204>.